### PR TITLE
Add Ruby 3.1/3.2 support

### PR DESCRIPTION
**Pull request by bastelfreak**
contains:
* https://github.com/voxpupuli/beaker_puppet_helpers/pull/4
* https://github.com/voxpupuli/beaker_puppet_helpers/pull/6
* https://github.com/voxpupuli/beaker_puppet_helpers/pull/7

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# on 2023-03-31 12:24:28 UTC using RuboCop version 1.48.1.
+# The point is for the user to remove these configuration records
+# one by one as the offenses are removed from the code base.
+# Note that changes in the inspected code, or installation of new
+# versions of RuboCop, may require this file to be generated again.
+
+# Offense count: 1
+# Configuration parameters: Severity, Include.
+# Include: **/*.gemspec
+Gemspec/RequiredRubyVersion:
```
Comment (ekohl) [inline .rubocop_todo.yml:12]: We have `required_ruby_version` in the gemspec, so why is this a problem?

**Review by ekohl — COMMENTED** [review]
Bleh, the Ruby 3 keyword arguments are biting. I kept `apply_manifest_on` compatible with beaker-puppet, but the last argument is interpreted differently on Ruby 3.

I'll think about this one for a bit, but now have to complete a separate migration.

File: .rubocop_todo.yml (hunk 1 above)
Comment (bastelfreak) [reply .rubocop_todo.yml:12]:
```
beaker_puppet_helpers.gemspec:13:29: W: Gemspec/RequiredRubyVersion: required_ruby_version and TargetRubyVersion (2.7, which may be specified in .rubocop.yml) should be equal.
  s.required_ruby_version = '>= 2.5', '< 4'
                            ^^^^^^^^^^^^^^^
```
I will fix that. rubocop was linting on a different version than the gemspec lists

**Comment by bastelfreak** [comment]
this PR is obsolete now

**Pull request closed without merging**